### ISSUE TITLE
UglifyJs: Unexpected token: name (listeners)

### ISSUE THREAD

**Issue by darkowic** [vesparny/brcast#7]
From webpack:
```
ERROR in vendor.8f839b5a1ada59daa303.js from UglifyJs
Unexpected token: name (listeners) [vendor.8f839b5a1ada59daa303.js:2174,6]
```
Caused because of this library (used in material-ui 1.0.0-beta.4).
After removing `"jsnext:main": "index.js",` from the library `package.json` build was done correctly

Webpack bundle:
```
/***/ "./node_modules/brcast/index.js":
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
Object.defineProperty(__webpack_exports__, "__esModule", { value: true });
/* harmony export (immutable) */ __webpack_exports__["default"] = createBroadcast;
function createBroadcast (initialState) {
  let listeners = {}
  let id = 0
  let _state = initialState

  const getState = () => _state

  const setState = state => {
    _state = state
    const keys = Object.keys(listeners)
    for (let i = 0; i < keys.length; i += 1) {
      // if a listener gets unsubscribed during setState we just skip it
      if (typeof listeners[keys[i]] !== 'undefined') {
        listeners[keys[i]](state)
      }
    }
  }

  const subscribe = listener => {
    if (typeof listener !== 'function') { throw new Error('listener must be a function.') }
    const currentId = id
    let isSubscribed = true
    listeners[currentId] = listener
    id += 1
    return function unsubscribe () {
      // in case unsubscribe gets called multiple times we simply return
      if (!isSubscribed) return
      isSubscribed = false
      delete listeners[currentId]
    }
  }

  return { getState, setState, subscribe }
}

/***/ }),
```
Error thrown in line `  let listeners = {}`.

Maybe my weback configuration is incorrect but I use a lot of different libraries and don't have any problem.

I had same issue few months ago in another library - here is interesting post
https://github.com/shuhei/material-colors/issues/13#issuecomment-270462099

**Comment by vesparny** [comment]
Yeah I see.
Problem is the `jsnext:main` points to the wrong file.
It should point to the same file as `module`.
I won't have access to my laptop for the next days but I'll fix it ASAP.

In the meantime you can either tell webpack to ignore jsnext or require explicitly the file you want from the dist folder

**Comment by oliviertassinari** [comment]
@jacobfunch is also experiencing errors with `let` not being transpiled. It should be good once `jsnext:main` point to the same value as `module`.

**Comment by vesparny** [comment]
Yes I know. I won't have access to my laptop until September.
Please require the file from `dist` folder in the meantime

**Comment by vesparny** [comment]
fixed in `brcast@3.0.1`

**Comment by oliviertassinari** [comment]
Thank you :).
